src/librustdoc/clean/mod.rs

```diff
@@ -1834,6 +1834,7 @@ pub(crate) fn clean_ty<'tcx>(ty: &hir::Ty<'tcx>, cx: &mut DocContext<'tcx>) -> T
         TyKind::UnsafeBinder(unsafe_binder_ty) => {
             UnsafeBinder(Box::new(clean_unsafe_binder_ty(unsafe_binder_ty, cx)))
         }
+        TyKind::FieldOf(..) => todo!("FIXME(field_projections): no idea what to do here"),
```

Comment:
Hmm it depends on how we will present this time on rustdoc. We could add a type variant in `src/librustdoc/clean/types.rs` without any additional information and defer as a future work. I am afraid that a wave of bikeshedding is inevitable.